lib/tokenize.cpp

```diff
@@ -822,8 +822,8 @@ namespace {
             }
 
             // don't add class|struct|union in inheritance list
-            if (Token::Match(tok->previous(), "public|private|protected") && Token::Match(mRangeType.first, "class|struct|union"))
-                mRangeType.first = mRangeType.first->next();
+            if (Token::Match(tok->previous(), "public|private|protected") && Token::Match(mRangeType.first, "const| class|struct|union"))
+                mRangeType.first = mRangeType.first->tokAt(mRangeType.first->str() == "const" ? 2 : 1);
```

Comment:
this removes the "const" that is not very good.

Imagine:

```
struct B {};
typedef const struct B CB;\n"
namespace N {
    struct D : public CB {};
}
CB cb; // <- This should be const!
```

What do you think about something like this:
```
    auto rangeType = mRangeType;
    if (Token::Match(tok->previous(), "public|private|protected")) {
        while (Token::Match(rangeType.first, "const|class|struct|union"))
            rangeType.first = rangeType.first.next();
    }

    Token* const tok2 = insertTokens(tok, rangeType);
```